LibZone/LibZone.lua

```diff
@@ -220,13 +383,18 @@ function lib:GetAllZoneDataById(reBuildNew, doReloadUI)
                 if zoneDataForId.zoneIndex == nil then
                     zoneDataForId.zoneIndex = zoneIndexOfZoneId
                 end
+				-- Get parentZoneId and pinInfo.
+				-- The parentZoneId is based on geographical location.
+				-- pinInfo is table of map pin indices {poiIndex = number, poiZoneIndex = number}
+				local parentZoneId, pinInfo = getPinInfo(zoneId, GetParentZoneId(zoneId))
                 --Set zone parent
-                if GetParentZoneId ~= nil then --> Function will be added with API100025 Murkmire
-                    if zoneDataForId.parentZone == nil then
-                        local zoneParentIdOfZoneId = GetParentZoneId(zoneId)
-                        zoneDataForId.parentZone = zoneParentIdOfZoneId
-                    end
-                end
+				if zoneDataForId.parentZone == nil then

```

Comment:
This can be ignored and the poiZoneIndex can be use for correct zone.

Reply:
In my current LibZone sv 1345 is the highest zoneId.
My original thought was that if an addon needed to, for example, check for summerset from Artaeum then they can GetParentZoneId(zoneId)
I'm going to put more thought into the parent.

Reply:
Looked good so far, but I did not understand all yet :-) Thanks for your contribution!
Currently too busy o do anything with ESO and addons...

One more hint: If you have moved the data to the Libone_Data.lua files add a local pointing to it within LibZone.lua (e.g. local wayshrineName = lib.WayshrineName so that the usage is not always checking the whole lib table for "wayshrineName" on each loop call. Same for the preloaded data, if possible.

As DanBatson said as an advice:
```
my general advice for locals/globals is that it's better to make a local of the global if you plan on indexing the global more than once, because global lookup is one of the more expesnive lookups. Expesnsive in this case is still mostly consequential, so you'll never likely notice a difference. But I try to minimize repeat global lookups of the same variable when I can. As for further indexing something inside the table, my general rule of thumb is if I'm going to index that same value more than 3 times, I'll make a local for it. so rather than do:

if obj.val and objv.val < 100 and obj.val > 10 then
I will do

local val = obj.val
if val and val < 100 and val > 10 then
```